### ISSUE TITLE
Reboot Docker host from a container

### ISSUE THREAD

**Issue by jfusterm** [docker/docker#21929]
While testing the Linux capabilities and seccomp options in Docker, and even with the `reboot` syscall and `CAP_SYS_BOOT` capability blocked by default ([1](https://github.com/docker/docker/blob/master/profiles/seccomp/default.json) & [2](https://github.com/docker/docker/blob/87de5fdd5972343a11847922e0f41d9898b5cff7/daemon/execdriver/native/template/default_template_linux.go#L16-L29)), I was able to reboot the Docker host from a container.

**Output of `docker version`:**

```
Client:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:59:07 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:59:07 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 0
 Running: 0
 Paused: 0
 Stopped: 0
Images: 1
Server Version: 1.10.3
Storage Driver: devicemapper
 Pool Name: docker-8:1-4326381-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 534.4 MB
 Data Space Total: 107.4 GB
 Data Space Available: 90.04 GB
 Metadata Space Used: 950.3 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/165536.165536/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/165536.165536/devicemapper/devicemapper/metadata
 Library Version: 1.02.99 (2015-06-20)
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 4.2.0-30-generic
Operating System: Ubuntu 15.10
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 984.2 MiB
Name: docker-test
ID: 3ES5:S6KH:5BYS:6GG2:RFB6:NTOM:O5OE:LL4E:ILKY:NTHC:NBQF:BMUV
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
```

**Additional environment details:**

The Docker host is an Ubuntu 15.10 virtual machine.

**Steps to reproduce the issue:**

1. Launch a container with systemd on it and bind-mount `/var/run/systemd`: `docker run -it --rm -v /var/run/systemd:/var/run/systemd debian:jessie`
2. Reboot the Docker host from the container: `root@a0ebd1a65c9b:/# systemctl reboot`

**Describe the results you received:**

Using `debian:jessie`, `ubuntu:15:10` or `centos:7` the result is the same, the Docker host got rebooted.

**Describe the results you expected:**

The expected result was an **"Operation not permitted"**.

**Additional information:**

1. The `-v /var/run/systemd:/var/run/systemd` is necessary to reboot the host.
2. Passing the option `--userns-remap=default` to the Docker daemon prevents the host reboot.
3. The same configuration was tested in a CentOS 7 Docker host with the same outcome.
4. Although the reboot is implicitly denied, denying it explicitly doesn't prevents the reboot.

```
# cat no_reboot.json

{
  "defaultAction": "SCMP_ACT_ALLOW",
  "syscalls": [
    {
      "name": "reboot",
      "action": "SCMP_ACT_ERRNO",
      "args": []
    }
  ]
}

# docker run -it --rm -v /var/run/systemd:/var/run/systemd --cap-drop SYS_BOOT --security-opt seccomp:no_reboot.json centos:7
[root@52969a8a25af /]# systemctl reboot
```

**Comment by thaJeztah** [comment]
Perhaps I'm mistaken, but I think this is expected; by using the volume, you're actually giving the container permissions to talk to the systemd daemon outside the container's namespace. This process is not controlled by the seccomp profile, thus can still perform this action. Similar to using `-v /var/run/docker.sock:/var/run/docker.sock`

**Comment by justincormack** [comment]
Yes that is correct. The `systemd` process is completely outside the control of docker, and you have allowed yourself a control channel to it that it will follow any instructions on. This is different from trying to run the actual reboot call from inside the container. You should not share the `systemd` socket if you do not want a process to be able to talk to it and start and stop services, reboot the machine etc.

**Comment by cpuguy83** [comment]
You can use an authorization plugin to restrict access to things like the systemd socket.
https://docs.docker.com/engine/extend/authorization/

**Comment by walkerleigh** [comment]
This "bug" got me out of trouble today, six years later. Something went wrong on my home server, ssh and VNC were refusing to let me in but Portainer was still running and accessible. So, I used Portainer to duplicate an existing container and mapped a volume as above, connected into the console again from Portainer then issued the systemctl reboot command and was back up and running a few minutes later. I am 10,500 milles away from the machine at the moment so very pleased to have found this.
